proposals/buildfile.md

```diff
@@ -0,0 +1,186 @@
+# Jib CLI Buildfile Specification
+
+Specification for a YAML buildfile describing building a container image. This buildfile can be
+used by the jib-cli to generate a container. It is translated directly into a buildplan and
+passed to the builder.
+
+```yaml
+apiVersion: jib/v0alpha1
+kind: Config
+
+# "FROM" with detail for manifest lists or multiple architectures
+from:
+  image: "gcr.io/distroless/java:8"
+  platforms:
+    - architecture: "arm"
+      os: "linux"
+      os.version: "a"
+      os.features:
+        - "b1"
+        - "b2"
+      variant: "c"
+      features:
+        - "d1"
+        - "d2"
+    - architecutre: amd64
+      os: darwin
+
+# potentially simple form of "FROM" (based on ability to define schema)
+from: "gcr.io/distroless/java:8"
+
+creationTime: 0 # millis since epoch or iso8601 creation time
+format: Docker # Docker or OCI
+
+environment:
+  "MY_KEY1": "value"
+  "MY_KEY2": "value"
+labels:
+  "com.example.owner": "person"
+  "com.example.mode": "dev"
+volumes:
+  - "/myvolume"
+  - "/youvolume"
+
+exposedPorts:
+  - "123/udp"
+  - "456"
+  - "789/tcp"
+
+user: root
+workingDirectory: "/somewhere"
+entrypoint:
+  - "java"
+  - "-jar"
+cmd:
+  - "myjar.jar"
+
+layers:
+  properties:
+    # file properties applied to all layers
+    filePermissions: "644"
+    directoryPermissions: "755"
+    user: "0"
+    group: "0"
+    timestamp: "0"
+  entries:
+    - name: "scripts and classes"
+      # file properties only applied to this layer "scripts and classes"
+      properties:
+        filePermissions: "333"
+        directoryPermissions: "777"
+        user: "goose"
+        group: "3"
+        timestamp: "2020-06-03T19:31:50+00:00"
+      files:
+        - from: "target/scripts"
+          to: "/app/scripts"
+          # file properties only applied to this copy directive
+          properties:
+            filePermissions: "777"
+          # another copy for the same layer, with includes and excludes
+        - from: "target/classes"
+          to: "/app/classes"
+          excludes:
+            - "**/goose.class"
+            - "**/moose.class"
+          includes:
+            - "**/*.class"
+
+      # another layer, only globally defined file permissions are applied here
+    - name: "other"
+      files:
+        - from: "build/other"
```

Comment:
I was thrown off by this: I thought the `from` was referring to a base image.  WDYT about `source` and `destination`?

Reply:
src, dest makes sense